upgrade-job/src/common/error.rs

```diff
@@ -0,0 +1,249 @@
+        pod_name,
+        pod_namespace,
+        source
+    ))]
+    GetPod {
+        source: kube::Error,
+        pod_name: String,
+        pod_namespace: String,
+    },
+
+    #[snafu(display(
+        "Failed to list Pods with label {} in namespace {}: {}",
+        label,
+        namespace,
+        source
+    ))]
+    ListPodsWithLabel {
+        source: kube::Error,
+        label: String,
+        namespace: String,
+    },
+
+    #[snafu(display("Failed get .spec from Pod {} in Namespace {}", name, namespace))]
+    EmptyPodSpec { name: String, namespace: String },
+
+    #[snafu(display(
+        "Failed get .spec.nodeName from Pod {} in Namespace {}",
+        name,
+        namespace
+    ))]
+    EmptyPodNodeName { name: String, namespace: String },
+
+    #[snafu(display("Failed to uncordon {} Node {}: {}", PRODUCT, node_name, source))]
+    StorageNodeUncordon {
+        source: openapi::tower::client::Error<openapi::models::RestJsonError>,
+        node_name: String,
+    },
+
+    #[snafu(display("Failed get delete Pod {} from Node {}: {}", name, node, source))]
+    PodDelete {
+        source: kube::Error,
+        name: String,
+        node: String,
+    },
+
+    #[snafu(display("Failed to list {} Nodes: {}", PRODUCT, source))]
+    ListStorageNodes {
+        source: openapi::tower::client::Error<openapi::models::RestJsonError>,
+    },
+
+    #[snafu(display("Failed to list {} Node {}: {}", PRODUCT, node_name, source))]
+    GetStorageNode {
+        source: openapi::tower::client::Error<openapi::models::RestJsonError>,
+        node_name: String,
+    },
+
+    #[snafu(display("Failed to get {} Node {}", PRODUCT, node_id))]
+    EmptyStorageNodeSpec { node_id: String },
+
+    #[snafu(display("Failed to list {} Volumes: {}", PRODUCT, source))]
+    ListStorageVolumes {
+        source: openapi::tower::client::Error<openapi::models::RestJsonError>,
+    },
+
+    #[snafu(display("Failed to drain {} Node {}: {}", PRODUCT, node_name, source))]
+    DrainStorageNode {
+        source: openapi::tower::client::Error<openapi::models::RestJsonError>,
+        node_name: String,
+    },
+
+    #[snafu(display("Pod {} in Namespace {} is not running", name, namespace))]
+    ValidatingPodRunningStatus { name: String, namespace: String },
+
+    #[snafu(display("Failed to parse YAML path {}", yaml_path))]
+    YamlStructure { yaml_path: String },
+
+    /// Error for use when converting Vec<> to String.
+    #[snafu(display("Failed to convert Vec<u8> to UTF-8 formatted String: {}", source))]
+    U8VectorToString { source: std::str::Utf8Error },
+
+    #[snafu(display("Failed to publish error: {}", source))]
+    EventPublish { source: kube_client::Error },
+}
+
+pub(crate) type Result<T, E = Error> = std::result::Result<T, E>;
+
+pub(crate) fn must<T>(output: Result<T>) -> T {
+    if let Err(error) = output {
+        tracing::error!(?error, "Failed to upgrade");
+        std::process::exit(-1);
+    }
+    output.unwrap()
```

Comment:
```suggestion
    match output {
        Err(error) => {
            tracing::error!(?error, "Failed to upgrade");
            std::process::exit(-1);
        }
        Ok(output) => output,
    }
```

Reply:
Why no propagate error rather than exit?

Reply:
Because I wanted the main function to be the highest error propagation. But, it makes sense to let tokio runtime handle the exiting.

Reply:
Done away with it, let me know what you think about this change.